nemo/collections/nlp/data/language_modeling/megatron/data_samplers.py

```diff
@@ -149,9 +150,13 @@ def __init__(
             pad_samples_to_global_batch_size == False
         ), "`MegatronPretrainingRandomSampler` does not support sample padding"
         self.last_batch_size = self.total_samples % self.micro_batch_times_data_parallel_size
+        self.seed = seed
 
     def __len__(self):
-        num_available_samples: int = self.total_samples
+        active_total_samples = self.total_samples - (self.last_batch_size if self.drop_last else 0)
+        num_available_samples = (
+            active_total_samples * (1 + (self.consumed_samples // active_total_samples))
+        ) - self.consumed_samples
```

Comment:
I find this formula hard to understand, while the following is more readable (IMO):
```
num_available_samples = active_total_samples - self.consumed_samples % active_total_samples
```

Reply:
Done, thanks